food-truck-api/src/main/java/food/truck/api/notification/NotificationService.java

```diff
@@ -99,4 +141,17 @@ public void saveNotification(Notification notification) {
     public void deleteNotification(Notification notification) {
         notificationRepository.delete(notification);
     }
+
+    public void saveNearbyNotification(Truck truck, String message) {
+        var routeLocation = truckService.getCurrentRouteLocation(truck.getId());
+        routeLocation.ifPresent(loc -> {
+            var n = new NearbyNotification();
+            n.setTruck(truck);
+            n.setMessage(message);
+            n.setTime(Instant.now());
+            n.setLatitude(loc.getPosition().getLatitude());
+            n.setLongitude(loc.getPosition().getLongitude());
+            nearbyNotificationRepository.save(n);
+        });
```

Comment:
I would make it send a 400 Bad Request or something if it is not present

Reply:
I'm not sure this is a great idea since we're also sending subscriber notifications. If there isn't an active route there isn't really a way to tell if there are nearby users, and so it just doesn't save it at all.